<?xml version="1.0"?>

<launch>

 <!-- Run box filter to remove car -->
 <node pkg="filtering_snow" type="cropBoxFilter" name="cropbox1" output="screen">
   <param name="inputTopic" value="/velodyne_points" type="string" />
   <rosparam param="maxvector">[2.5, 0.935, 0]</rosparam>
   <rosparam param="minvector">[-2.5, -0.935, -1.480]</rosparam>
   <param name="setNegative" value="true" type="bool" />
   <param name="outputNoPoints" value="false" type="bool" />
 </node>

 <!-- Run box filter to remove far features and ground -->
 <node pkg="filtering_snow" type="cropBoxFilter" name="cropbox2" output="screen">
   <param name="inputTopic" value="/cropbox1/output" type="string" />
   <rosparam param="maxvector">[8, 8, 20]</rosparam>
   <rosparam param="minvector">[-8, -8, -1.300]</rosparam>
   <param name="setNegative" value="false" type="bool" />
   <param name="outputNoPoints" value="true" type="bool" />
 </node>

  <!-- Run dynamic radius outlier removal filter
        - set the min. no. of neighbours to be within the search radius for the point to be included
        - Set the radius search multiplier (i.e. radius_search = radius_multiplier * azAngle * range)
        - Set the azimuth angle of the lidar (in degrees, for VLP16 - 0.4)
        - Set the minimum search radius (default 0) This stops all the close points from being removed
        - Note: you need to specify a search radius because the dynamic search radius
                only works for PointCloud2 right now. If PointCloud is used, it
                will use the specified static search range-->

  <node pkg="filtering_snow" type="dynamicRadiusOutlierFilter" name="dynamic_radius_outlier_filter" output="screen"/>
    <param name="/radiusDynamic/radius_search" value="0.8" type="double" />
    <param name="/radiusDynamic/inputTopic" value="/velodyne_points" type="string" />
    <param name="/radiusDynamic/min_Neighbours" value="3" type="int" />
    <param name="/radiusDynamic/radius_multiplier" value="1.1" type="double" />
    <param name="/radiusDynamic/azimuth_angle" value="0.4" type="double" />
    <param name="/radiusDynamic/min_search_radius" value="0.03" type="double" />

  <!-- Run radius outlier removal filter
     - set the radius to search around each point
     - set the min. no. of neighbours to be within the search radius for the point to be included -->
  <node pkg="filtering_snow" type="radiusOutlierFilter" name="radius_outlier_filter" output="screen"/>
    <param name="/radius/inputTopic" value="/velodyne_points" type="string" />
    <param name="/radius/radius_search" value="0.05" type="double" />
    <param name="/radius/minNeighbours" value="3" type="int" />

  <!-- Run rviz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find filtering_snow)/config/rviz_config.rviz" />

</launch>
